Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

Support the new location for the acpica version information, as /proc/acpi/info
no longers exists in 2.6.21 development tree.

diff -Nuar acpitool-0.4.7.orig/src/acpitool.cpp acpitool-0.4.7/src/acpitool.cpp
--- acpitool-0.4.7.orig/src/acpitool.cpp	2007-02-06 12:03:02.000000000 -0800
+++ acpitool-0.4.7/src/acpitool.cpp	2007-02-12 12:04:35.000000000 -0800
@@ -54,15 +54,18 @@
     char *filename;
     
     filename = "/proc/acpi/info";
-    
     file_in.open(filename);
-    if (!file_in)
-    	return 0;
-    else
-    {
+    if(file_in) goto has_success;
+    filename = "/sys/module/acpi/parameters/acpica_version";
+    file_in.open(filename);
+    if(file_in) goto has_success;
+    // failure
+    return 0;
+
+has_success:
 	file_in.close();
         return 1;
-    }
+
 }
     
 int Print_ACPI_Info(int show_ac, int show_therm, int show_trip, int show_fan, int show_batteries, int show_empty, int show_version, int show_cpu, int show_wake, int e_set, int info_level, int verbose)
@@ -96,23 +99,27 @@
     char *filename, str[50];
     
     filename = "/proc/acpi/info";
-    
     file_in.open(filename);
-    if (!file_in)
+    if(file_in) goto version_success;
+    filename = "/sys/module/acpi/parameters/acpica_version";
+    file_in.open(filename);
+    if(file_in) goto version_success;
+
+    // failure
+    if(!verbose)
     {
-    	if(!verbose)
-    	{
 	    strcpy(c, "<n.a>");
-    	    return 0;
-    	}
-    	else
-    	{
-    	    cout<<"  Could not open file : "<<filename<<endl;
+	    return 0;
+    }
+    else
+    {
+	    cout<<"  Could not open file : "<<filename<<endl;
 	    cout<<"  Make sure your kernel has ACPI support enabled."<<endl;
 	    return -1;
-    	}	
-    }
-       
+    }	
+
+version_success:
+
     file_in.getline(str, 50);
     file_in.close();
     
